features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/ApiSecurity.jsx

```diff
@@ -0,0 +1,495 @@
+import Tooltip from '@material-ui/core/Tooltip';
+import HelpOutline from '@material-ui/icons/HelpOutline';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+const styles = theme => ({
+    titleWrapper: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        paddingBottom: 20,
+    },
+    root: {
+        ...theme.mixins.gutters(),
+        paddingTop: theme.spacing.unit * 2,
+        paddingBottom: theme.spacing.unit * 2,
+    },
+    checkItem: {
+        textAlign: 'center',
+    },
+    divider: {
+        marginTop: 20,
+        marginBottom: 20,
+    },
+    chip: {
+        margin: theme.spacing.unit / 2,
+        padding: 0,
+        height: 'auto',
+        '& span': {
+            padding: '0 5px',
+        },
+    },
+    imageContainer: {
+        display: 'flex',
+    },
+    imageWrapper: {
+        marginRight: theme.spacing.unit * 3,
+        width: 200,
+    },
+    subtitle: {
+        marginTop: theme.spacing.unit,
+    },
+    specialGap: {
+        marginTop: theme.spacing.unit * 3,
+    },
+    resourceTitle: {
+        marginBottom: theme.spacing.unit * 3,
+    },
+    ListRoot: {
+        padding: 0,
+        margin: 0,
+    },
+    title: {
+        flex: 1,
+    },
+    helpButton: {
+        padding: 0,
+        minWidth: 20,
+    },
+    helpIcon: {
+        fontSize: 16,
+    },
+    htmlTooltip: {
+        backgroundColor: '#f5f5f9',
+        color: 'rgba(0, 0, 0, 0.87)',
+        maxWidth: 220,
+        fontSize: theme.typography.pxToRem(14),
+        border: '1px solid #dadde9',
+        '& b': {
+            fontWeight: theme.typography.fontWeightMedium,
+        },
+    },
+    buttonWrapper: {
+        paddingTop: 20,
+    },
+    descriptionTextField: {
+        width: '100%',
+        marginBottom: 20,
+    },
+    rightDataColum: {
+        flex: 1,
+    },
+    formControlLeft: {
+        width: 100,
+    },
+    formControlRight: {
+        flex: 1,
+    },
+    inlineForms: {
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'flex-end',
+        paddingBottom: 12,
+    },
+    textFieldRoles: {
+        padding: 0,
+        margin: '0 0 0 10px',
+    },
+    group: {
+        flexDirection: 'row',
+    },
+    error: {
+        lineHeight: '31px',
+    },
+    authFormControl: {
+        marginTop: 0,
+    },
+});
+
+/**
+ * The component for api security configurations.
+ * @param {any} props The input props.
+ * @returns {any} The HTML representation of the API Security.
+ */
+function ApiSecurity(props) {
+    const {
+        classes, api, isTransportHttps, securityScheme, error, setSecurityScheme, addToArray, removeFromArray,
+        securitySchemaValues,
+    } = props;
+
+    const getAPISecurityState = (type, apiSecurityScheme) => {
```

Comment:
It's better if we can add method-level comments.

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/commit/636f93611578bba36bc677fb4eb1bcdf281489a5